report/record.go

```diff
@@ -0,0 +1,92 @@
+package report
+
+import (
+	"io"
+	"sync"
+
+	"github.com/pterm/pterm"
+)
+
+var m = &sync.Mutex{}
+
+var records = make(map[string]Entry)
```

Comment:
Hey @zackproser, have we considered `NukeAllResources` method returning the status of each `nuke` operation, rather than using global + mutex pattern? 

I can think of one benefit as having full understanding of what the `NukeAllResources` method does without worrying about side outputs it produce via global variables.

Reply:
Ah interesting. I hadn't considered that but you could well be right about the approach. Could you please elaborate a bit more on how that would work to ensure I understand?

Is your chief concern the use of the `m` and `records` variables within this package? They should only be usable by methods within this package to prevent race conditions.